BST put more emphasize that balanced (#29580)
TheOdinProject/curriculum: javascript/computer_science/project_binary_search_trees.md

1. Write a `depth(node)` function that returns the given node's depth. Depth is defined as the number of edges in the path from a given node to the tree's root node.

> 1. Write a `isBalanced` method that checks if the tree is balanced. A binary tree is considered balanced if, for every node in the tree, the height difference between its left and right subtrees is no more than 1, and both the left and right subtrees are also balanced.

Reviewer: ```suggestion
1. Write a `isBalanced` method that checks if the tree is balanced. A binary tree is considered balanced if, for every node in the tree, the height difference between its left and right subtrees is no more than 1, and both the left and right subtrees are also balanced.
```
```suggestion
1. Write an `isBalanced` function that checks if the tree is balanced. A binary tree is considered balanced if, for every node in the tree, the height difference between its left and right subtrees is no more than 1, and both the left and right subtrees are also balanced.
```
Small nitpick here on English grammar. Needs to be "an `isBalanced`" instead of "a `isBalanced`" due to the first syllable in `isBalanced` being a vowel sound.

Also just to stay consistent with the other instructions in the JavaScript course's version of the project, I'm going to recommend using "function" instead of "method."

Author: Resolved. Thanks!

<div class="lesson-note lesson-note--tip" markdown="1">

#### Pitfall with checking balance

> A common mistake is only checking the height difference between the root's left and right children. That is not enough — you must check the balance condition recursively at every node.

Reviewer 2: @JoshDevHub thoughts on the suggestion below? I forsee people misinterpreting it to mean that they *must* solve this with recursion.

```suggestion
A common mistake is only checking the height difference between the root's left and right children. That is not enough — you must check the balance condition for every node.
```

Reviewer: Recursion is definitely the way you'd want to do this task, but I'm also fine leaving this more to the learner to discover for themselves.

So yeah, I support it. @ChuqiaoHuang would you be okay to implement this change and the one below? Then we can merge.

Author: Yes, I have changed it. @JoshDevHub
